portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -12,21 +12,75 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package com.google.sps.servlets;
-
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.util.ArrayList;
+import java.util.List;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+package com.google.sps.servlets;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> quotes;
+
+  @Override
+  public void init() {
+    quotes = new ArrayList<>();
+    quotes.add("You must be the change you wish to see in the world. - Gandhi");
+    quotes.add("Creativity is intelligence having fun. - Albert Einstein");
+    quotes.add(
+        "Whether you think you can or you think you can't, "
+            + "you're right. - Henry Ford");
+    quotes.add(
+        "Go confidently in the direction of your dreams. Live the life you have imagined. "
+            + "- Henry David Thoreau");
+    quotes.add("Everything has beauty, but not everyone can see. - Confucius");
+    quotes.add("Impossible is nothing. - Muhammad Ali");
+    quotes.add("Don't count the days, make the days count. - Muhammad Ali");
+    quotes.add(
+        "Darkness cannot drive out darkness; only light can do that. "
+            + "Hate cannot drive out hate; only love can do that. - Martin Luther King Jr.");
+    quotes.add("The time is always right to do what is right. - Martin Luther King Jr.");
+  }
+
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String quote = quotes.get((int) (Math.random() * quotes.size()));
+
+    String json = convertToJSON(quote);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String userQuote = getParameter(request, "userQuote", "");
+    String userQuoteAuthor = getParameter(request, "userQuoteAuthor", "");
+
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jason</h1>");
+    response.getWriter().println(userQuote + " - " + userQuoteAuthor);
+  }  
+
+  private String convertToJSON(String quotes) {
+    GsonBuilder builder = new GsonBuilder();
+    builder.disableHtmlEscaping();
+    Gson gson = builder.create();
+    return gson.toJson(quotes);
+  }
+
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value === null) {
```

Comment:
instead of allowing null http params, you could investigate how to throw an exception instead. it sounds like the quote and author should never be null or an empty string